docs/book/misc/handling-dependencies.md

```diff
@@ -0,0 +1,80 @@
+conflicts if you are using other libraries in your environment. An easy way to
+see if the ZenML requirements are still met (after installing any extra
+dependencies required by your work) by running `zenml integration list` and
+checking that your desired integrations still bear the green tick symbol
+denoting that all requirements are met.
+
+## Suggestions for Resolving Dependency Conflicts
+
+### Use a tool like `pip-compile` for reproduciblity
+
+Consider using a tool like `pip-compile` (available through [the `pip-tools`
+package](https://pip-tools.readthedocs.io/)) to compile your dependencies into a
+static `requirements.txt` file that can be used across environments.
+
+For a practical example and explanation of using `pip-compile` to address
+exactly this need, see [our 'gitflow' repository and
+workflow](https://github.com/zenml-io/zenml-gitflow#-software-requirements-management)
+to learn more.
+
+### Use `pip check` to discover dependency conflicts
+
+Running [`pip check`](https://pip.pypa.io/en/stable/cli/pip_check/) will verify
+that your environment's dependencies are compatible with one another. If not,
+you will see a list of the conflicts. This may or may not be a problem or
+something that will prevent you from moving forward with your specific use case,
+but it is certainly worth being aware of whether this is the case.
+
+### Well-known dependency resolution issues
+
+Some of ZenML's integrations come with strict dependency and package version
+requirements. We try to keep these dependency requirements ranges as wide as
+possible for the integrations developed by ZenML, but it is not always possible
+to make this work completely smoothly. Here are some of the known issues:
+
+- `label_studio`: the packages required for this integration are not compatible
+  with quite a few of our other integrations. At a minimum, you will get
+  warnings when installing `label_studio` via `zenml integration install ...`.
+  We have found these warnings to be mostly harmless. You might find you need to
+  separate your installations of Label Studio and any other integrations you
+  need for your environment / pipeline, however. Note [the warning on our own
+  documentation
+  page](https://docs.zenml.io/component-gallery/annotators/label-studio#how-to-deploy-it)
+  for the Label Studio integration which currently breaks ZenML's CLI.
+- `click`: ZenML currently requires `click~=8.0.3` for its CLI. This is on
+  account of another dependency of ZenML. Using versions of `click` in your own
+  project that are greater than 8.0.3 may cause unanticipated behaviors.
+
+### Manually bypassing ZenML's integration installation
+
+It is possible to skip ZenML's integration installation process and install
+dependencies manually. This is not recommended, but it is possible and can be
+run at your own risk.
+
+To do this, you will need to install the dependencies for the integration you
```

Comment:
Maybe add here that `zenml integration install` essentially runs a `pip install ..` in the background and give them an example of an integration install like `zenml integration install gcp` vs a `pip install gcp-lib-1 gcp-lib-2` etc